### PR TITLE
Add nix build Action

### PR DESCRIPTION
**Pull request by butlerx**
Add Github action to build nixos on pr

uses https://github.com/marketplace/actions/install-nix

### CODE REVIEW

**Comment by m1cr0man** [comment]
It very likely won't work without all the secret files too, which are scattered across a lot of the service configs. These can be mocked.

**Comment by butlerx** [comment]
currently, this is failing but this pr is as good a place as any to track the work for setting this up.

**Comment by butlerx** [comment]
- [ ] `error: getting status of '/home/runner/work/nix-configs/nix-configs/default.nix': No such file or directory`
- [ ] `There are environment variables missing from this runtime, but would be present on GitHub. - GITHUB_TOKEN`

**Review by benmcmahon100 — APPROVED** [review]

File: generate_test_conf.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+do
+  echo "    $SERVICE"
+done
+
+echo "  ];
+  # This value determines the NixOS release with which your system is to be
+  # compatible, in order to avoid breaking some software such as database
+  # servers. You should change this only after NixOS release notes say you
+  # should.
+  system.stateVersion = "19.09";
```
Comment (butlerx) [inline generate_test_conf.sh]: Wrong version

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  pull_request:
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set WIP to Draft
+      uses: wip/action@v1.0.0
+    - uses: cachix/install-nix-action@v8
+    - run: bash ./generate_test_conf.sh > default.nix
```
Comment (butlerx) [inline .github/workflows/main.yml:15]: Id rather use nix rather then bash

**Review by m1cr0man — APPROVED** [review]
This looks good but increase the version to 20.09

**Pull request merged**